src/solutions/ode_solutions.jl

```diff
@@ -74,6 +74,44 @@ function (sol::ODESolution)(t::AbstractVector{<:Real},deriv,idxs::AbstractVector
   DiffEqArray([[interp_sol[idx][i] for idx in idxs] for i in 1:length(t)], t, idxs, getindepsym(sol), observed, p)
 end
 
+function interpolation_residual(sol::AbstractODESolution, t)
+  sol(t,Val{1}) - sol.prob.f(sol(t), sol.prob.p, t)
```

Comment:
Will this syntax work for the "inplace" syntax exemplified by the [Lorenz ODE example](https://diffeq.sciml.ai/stable/tutorials/ode_example/#Example-2:-Solving-Systems-of-Equations)?

Reply:
no, you'll want to check whether `f` isinplace and then write into a cache vector via `similar(u)`.